pkg/api/mock/matcher/common_test.go

```diff
@@ -12,13 +14,18 @@ type testObject struct {
 }
 
 func (o *testObject) EndpointURL(ctx context.Context) (*url.URL, error) { return nil, nil }
+func (o *testObject) GetIdentifier(context.Context) (string, error)     { return o.Identifier, nil }
 
 type testObject2 struct {
 	Identifier string `anxcloud:"identifier"`
 }
 
 func (o *testObject2) EndpointURL(ctx context.Context) (*url.URL, error) { return nil, nil }
+func (o *testObject2) GetIdentifier(context.Context) (string, error)     { return o.Identifier, nil }
 
 type testObjectNotStruct string
 
 func (o *testObjectNotStruct) EndpointURL(ctx context.Context) (*url.URL, error) { return nil, nil }
+func (o *testObjectNotStruct) GetIdentifier(context.Context) (string, error) {
+	return "", types.ErrObjectWithoutIdentifier
```

Comment:
the purpose of this test object is to check how we handle non-struct Objects - please return the string itself as identifier.

Maybe we have to add a "is struct" check somewhere in generic client or strip the requirement of Objects being structs - which was only in place to find the Identifier attribute in them. IIRC there is outside code relying on them being structs though.. not sure what to do, yet.